libraries/chain/include/eosio/chain/symbol.hpp

```diff
@@ -0,0 +1,151 @@
+/**
+ *  @file
+ *  @copyright defined in eos/LICENSE.txt
+ */
+#pragma once
+#include <fc/exception/exception.hpp>
+#include <eosio/chain/types.hpp>
+#include <string>
+
+namespace eosio {
+   namespace chain {
+      
+      static constexpr uint64_t string_to_symbol_c(uint8_t precision, const char* str) {
+         uint32_t len = 0;
+         while (str[len]) ++len;
+         
+         uint64_t result = 0;
+         // No validation is done at compile time
+         for (uint32_t i = 0; i < len; ++i) {
+            result |= (uint64_t(str[i]) << (8*(1+i)));
+         }
+
+         result |= uint64_t(precision);
+         return result;
+      }
+      
+#define SY(P,X) ::eosio::chain::string_to_symbol_c(P,#X)
+
+      static uint64_t string_to_symbol(uint8_t precision, const char* str) {
+         try {
+            uint32_t len = 0;
+            while(str[len]) ++len;
+            uint64_t result = 0;
+            for (uint32_t i = 0; i < len; ++i) {
+               // All characters must be upper case alaphabets
+               FC_ASSERT (str[i] >= 'A' && str[i] <= 'Z', "invalid character in symbol name");
+               result |= (uint64_t(str[i]) << (8*(i+1)));
+            }
+            result |= uint64_t(precision);
+            return result;
+         } FC_CAPTURE_LOG_AND_RETHROW((str))
+      }
+
+      class symbol {
+         public:
+            symbol(uint8_t p, const char* s): m_value(string_to_symbol(p, s)) { }
+            symbol(uint64_t v = SY(4, EOS)): m_value(v) { }
```

Comment:
Add `explicit`

Reply:
Done. I fixed the parts of the code that this breaks.